### PR TITLE
fix: visual viewport types and body check

### PR DESCRIPTION
**Pull request by atomiks**

### CODE REVIEW

**Comment by FezVrasta** [comment]
I just fixed the Flow issues on master, may you rebase pls?

File: src/dom-utils/listScrollParents.js [hunk 1]
```diff
@@ -5,15 +5,25 @@ import getNodeName from './getNodeName';
...
+type VisualViewport = {
+  width: number,
+  height: number,
+  offsetLeft: number,
+  offsetTop: number,
+};
+
 export default function listScrollParents(
   element: Node,
+  list: Array<Element | Window | VisualViewport> = []
```
Comment (FezVrasta) [inline src/dom-utils/listScrollParents.js:17]:
```suggestion
```

File: src/dom-utils/listScrollParents.js [hunk 2]
```diff
@@ -5,15 +5,25 @@ import getNodeName from './getNodeName';
 import getWindow from './getWindow';
 import type { Window, VisualViewport } from '../types';
 
+type VisualViewport = {
+  width: number,
+  height: number,
+  offsetLeft: number,
+  offsetTop: number,
+};
```
Comment (FezVrasta) [inline src/dom-utils/listScrollParents.js:13]:
```suggestion
```

File: src/dom-utils/listScrollParents.js [hunk 3]
```diff
@@ -7,13 +7,15 @@ import type { Window, VisualViewport } from '../types';
 
 export default function listScrollParents(
   element: Node,
-  list: Array<Element | Window> = []
+  list: Array<Element | Window | VisualViewport> = []
```
Comment (FezVrasta) [inline src/dom-utils/listScrollParents.js:10]: I don't think we need VisualViewport here, at least, without it everything still works fine.

Comment (atomiks) [reply src/dom-utils/listScrollParents.js]: okay I think it's finally ready 🤣

File: src/dom-utils/listScrollParents.js [hunk 4]
```diff
@@ -12,8 +12,10 @@ export default function listScrollParents(
   const scrollParent = getScrollParent(element);
   const isBody = getNodeName(scrollParent) === 'body';
   const win = getWindow(scrollParent);
-  const target = isBody ? win : scrollParent;
-  const updatedList = list.concat(target, win.visualViewport || []);
+  const target = isBody
+    ? [].concat(win, win.visualViewport || [])
```
Comment (FezVrasta) [inline src/dom-utils/listScrollParents.js:16]:
```suggestion
    ? [win].concat(win.visualViewport || [])
```
this is a bit shorter

**Pull request merged**